doc/man3/BIO_s_datagram.pod

```diff
@@ -0,0 +1,255 @@
+subsequent call to BIO_write() being set to the source address of the datagram
+received by that call to BIO_read(). Thus, unless the caller calls
+BIO_dgram_set_peer() immediately prior to every call to BIO_write(), or never
+calls BIO_read(), any host on the network may cause future datagrams written to
+be redirected to that host. Therefore, it is recommended that users either use
+BIO_s_dgram() only with a connected socket, or, if using BIO_s_dgram() with an
+unconnected socket, to use the L<BIO_sendmmsg(3)> and L<BIO_recvmmsg(3)> methods
+only and forego use of L<BIO_read(3)> and L<BIO_write(3)>. An exception is where
+L<DTLSv1_listen(3)> must be used; see L<DTLSv1_listen(3)> for further
+discussion.
+
+=item
+
+Unlike L<BIO_read(3)> and L<BIO_write(3)>, the L<BIO_sendmmsg(3)> and
+L<BIO_recvmmsg(3)> methods are stateless and do not cause the internal state of
+the BIO_s_datagram() to change.
+
+=back
+
+Various controls are available for configuring the BIO_s_datagram() using
+L<BIO_ctrl(3)>:
+
+=over 4
+
+=item BIO_ctrl_dgram_connect (BIO_CTRL_DGRAM_CONNECT)
+
+This is equivalent to calling L<BIO_dgram_set_peer(3)>.
+
+Despite its name, this function does not cause the underlying socket to become
+connected.
+
+=item BIO_ctrl_set_connected (BIO_CTRL_SET_CONNECTED)
+
+This informs the BIO_s_datagram() whether the underlying socket has been
+connected, and therefore how the BIO_s_datagram() should attempt to use the
+socket.
+
+If the peer argument is non-NULL, BIO_s_datagram() assumes that the underlying
+socket has been connected and will attempt to use the socket using OS APIs which
+do not specify peer addresses (for example, send(3) and recv(3) or
+similar). The peer argument should specify the peer address to which the socket
+is connected.
+
+If the peer argument is NULL, BIO_s_datagram() assumes that the underlying
+socket is not connected and will attempt to use the socket using an OS APIs
+which specify peer addresses (for example, sendto(3) and recvfrom(3)).
+
+This control does not affect the operation of L<BIO_sendmmsg(3)> or
+L<BIO_recvmmsg(3)>.
+
+=item BIO_dgram_get_peer (BIO_CTRL_DGRAM_GET_PEER)
+
+This outputs a B<BIO_ADDR> which specifies one of the following values,
+whichever happened most recently:
+
+=over 4
+
+=item
+
+The peer address last passed to BIO_dgram_set_peer(), BIO_ctrl_dgram_connect()
+or BIO_ctrl_set_connected().
+
+=item
+
+The peer address of the datagram last received by a call to BIO_read().
+
+=back
+
+=item BIO_dgram_set_peer (BIO_CTRL_DGRAM_SET_PEER)
+
+Sets the peer address to be used for subsequent writes to this BIO.
+
+Warning: When used with an unconnected network socket, the value set may be
+modified by future calls to L<BIO_read(3)>, making use of BIO_s_datagram()
+hazardous when used with unconnected network sockets; see above.
+
+This does not affect the operation of L<BIO_sendmmsg(3)>, and L<BIO_recvmmsg(3)>
+does not affect this value.
```

Comment:
Not sure what this is meant to be saying, but it doesn't parse.

Reply:
Reads OK to me?

Reply:
Maybe just split into two sentences to make it clear?

Reply:
Reading it again, I can parse it.  Bloody Oxford commas :disappointed:
I think it's better in twain.